### PR TITLE
Update cloud nodesource description

### PR DESCRIPTION
**Pull request by tao-xinxiu**
- Rename section 4 from `Installation on a Cluster` to `Deploy ProActive Nodes`.
- change the outline of section 4 (Deploy ProActive Nodes): organize the part of "deploying from ProActive Scheduler" depending on the Compute Hosts types.
- Add a general description of deploying ProActive Nodes on Cloud Platforms.
- Move the detailed parameters description of `NativeSchedulerInfrastructure` and `NativeSchedulerPolicy` from section 4 to the reference.
- Add "GCE Infrastructure" description in the reference

### CODE REVIEW

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 1]
```diff
@@ -172,33 +172,69 @@ then you have to download the database driver and put it at: *PROACTIVE_HOME/add
 
 Scheduler, and 3 microservices (*scheduling-api*, *job-planner*, and *proactive-cloud-watch*) have to point to the same database.
 
-== Installation on a Cluster
+== Deploy ProActive Nodes
 
-Adding Compute Hosts of a cluster to the ProActive Scheduler typically involves unpacking the release archive on all those hosts.
-Once it's done you need to run a ProActive Node on the Compute Host and connect it to the ProActive Scheduler. There are two principal ways of
-doing that:
+In the ProActive Scheduler, all the computational resources (<<_glossary_compute_host,Compute Hosts>>) are managed as a set of <<_glossary_node_source,*Node Sources*>>.
```
Comment (fviale) [inline src/docs/admin/ProActiveAdminGuide.adoc]: Computational Resources are managed by the ProActive Resource Manager component.

In the ProActive Resource Manager, ...

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 2]
```diff
@@ -172,33 +172,69 @@ then you have to download the database driver and put it at: *PROACTIVE_HOME/add
...
+[NOTE]
+====
+Examples of a Node Source:
+
+* a cluster with SSH access where nodes are available from 9 a.m. to 9 p.m.
+* nodes from Amazon EC2 available permanently for users from group 'cloud'.
+====
+
+image::rm-portal.png[align=center]
+As shown in the previous screenshot, the ProActive Scheduler provides a <<_glossary_rm_web_interface,*Resource Manager Web Interface*>> to show and manage all the node sources with its nodes.
```
Comment (fviale) [inline src/docs/admin/ProActiveAdminGuide.adoc]: the ProActive Server provides

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 3]
```diff
@@ -172,33 +172,69 @@ then you have to download the database driver and put it at: *PROACTIVE_HOME/add
...
+
+Adding a cluster of Compute Hosts to the ProActive Scheduler requires to *create a node source*.
```
Comment (fviale) [inline src/docs/admin/ProActiveAdminGuide.adoc]: to the ProActive Resource Manager ...

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 4]
```diff
@@ -172,33 +172,69 @@ then you have to download the database driver and put it at: *PROACTIVE_HOME/add
...
+Its typical process involves copying the necessary ProActive binary files to all those Compute Hosts,
+running ProActive Nodes on the hosts, and connecting the nodes to the ProActive Scheduler.
```
Comment (fviale) [inline src/docs/admin/ProActiveAdminGuide.adoc]: to the Resource Manager

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 5]
```diff
@@ -172,33 +172,69 @@ then you have to download the database driver and put it at: *PROACTIVE_HOME/add
...
+There are two principal ways of doing that:
+
+- Launch a process of ProActive Node *on the Compute Host* and connect it to the ProActive Scheduler
+- Initiate the deployment *from the ProActive Scheduler*
+
+If you are not familiar with ProActive Scheduler you may want to try the <<run_node_manually,*first method*>> as it's directly executed on the Compute Hosts.
```
Comment (fviale) [inline src/docs/admin/ProActiveAdminGuide.adoc]: with ProActive server configuration,

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 6]
```diff
@@ -172,33 +172,69 @@ then you have to download the database driver and put it at: *PROACTIVE_HOME/add
...
+Combined with <<_proactive_agents,*ProActive Agents*>> which control and limit the resources utilization on the Compute Hosts,
+it gives you the same result as the *second method*.
 
-* Launch a process on the Compute Host and connect it to the ProActive Scheduler
-* Initiate the deployment from the ProActive Scheduler: <<_deploy_nodes_via_ssh,Node Source creation>>
+The <<_deploy_nodes_from_proactive_scheduler,*second method*>> commands the ProActive Scheduler to remotely launch the process of ProActive Nodes on Compute Hosts.
```
Comment (fviale) [inline src/docs/admin/ProActiveAdminGuide.adoc]: commands ProActive Resource Manager

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 7]
```diff
@@ -172,33 +172,69 @@ then you have to download the database driver and put it at: *PROACTIVE_HOME/add
...
+The method implies that you have a remote access to Compute Hosts (e.g., SSH access).
 
-If you are not familiar with ProActive Scheduler you may want to try the *first method* as it's easier to understand.
-Combined with <<_proactive_agents,*ProActive Agents*>>, it gives you the same result as the *second method*.
+The ProActive Nodes can be deployed on various kinds of computational resources, such as:
 
-The *second method* implies that you have a remote access to Compute Hosts (e.g. SSH access) and you want to start and stop ProActive Nodes
-by launching commands remotely on Compute Hosts. For instance, it can be useful when a virtual machine needs to be deployed prior to launching a ProActive Node.
+- the local Server Host (i.e., the machine installed the ProActive Scheduler)
```
Comment (fviale) [inline src/docs/admin/ProActiveAdminGuide.adoc]: i.e., the machine where the ProActive Server is installed.

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 8]
```diff
@@ -172,33 +172,69 @@ then you have to download the database driver and put it at: *PROACTIVE_HOME/add
...
+- Cloud Platforms (e.g., Amazon Web Services, OpenStack, GoogleComputeEngine)
+- an existing cluster managed by another scheduler (e.g., SLURM, LSF, PBS)
+
+A node source is defined by two parts of configurations: <<_glossary_node_source_infrastructure,*Node Source Infrastructure*>> and <<_glossary_node_source_policy,*Node Source Policy*>>.
+Each *Node Source Infrastructure* defines the deployment mechanism for a specific kind of computational resources.
+To support the deployment of these various computational resources, the ProActive Scheduler provides a set of supported <<_node_source_infrastructure, Node Source Infrastructures>>.
+The *Node Source Policy* controls the resource usage. See the section <<_control_the_resource_usage>> for more details.
+This section concentrates on the deployment mechanism of ProActive Nodes which is defined by the Node Source Infrastructure.
+
+When you connect the Compute Hosts to the ProActive Scheduler through the *first method*, a node source of <<_default_infrastructure_manager,*Default Infrastructure*>> is automatically created.
```
Comment (fviale) [inline src/docs/admin/ProActiveAdminGuide.adoc]: to the Resource Manager through

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 9]
```diff
@@ -172,33 +172,69 @@ then you have to download the database driver and put it at: *PROACTIVE_HOME/add
...
+- using <<proactive_node,*proactive-node*>> command
+- using <<node_jar,*node.jar*>>
+- using <<proactive_node_autoupdate,*proactive-node-autoupdate*>> command
 
 [[proactive_node]]
 ==== Using proactive-node command
 
-Let's take a closer look at the first method described above. To deploy a ProActive Node from the Compute Host you need
-to run the following command
+To use `proactive-node` command, you need to unpack the release archive of the ProActive Scheduler on the Compute Host and run the following command
```
Comment (fviale) [inline src/docs/admin/ProActiveAdminGuide.adoc]: unpack the ProActive node release archive  inside the Compute Host and run the following command (server archive can also be used for this purpose although it is more appropriate to use the node archive) :

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 10]
```diff
@@ -172,33 +172,69 @@ then you have to download the database driver and put it at: *PROACTIVE_HOME/add
...
 ----
 $ PROACTIVE_HOME/bin/proactive-node -r pnp://SCHEDULER_ADDRESS:64738
 ----
-where `-r` option is used to specify the URL of the *Resource Manager*.
+where `-r` option is used to specify the URL of the <<_glossary_resource_manager,*Resource Manager*>>.
 You can find this URL in the output of the <<_run_the_proactive_scheduler,ProActive Scheduler>> (the Resource Manager URL).
```
Comment (fviale) [inline src/docs/admin/ProActiveAdminGuide.adoc]: <<_run_the_proactive_scheduler,ProActive Server command>>

Please add a sample output of the server command describing which url is requested (people often make mistakes and use http urls)

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 11]
```diff
@@ -210,7 +246,7 @@ TIP: You can also use *discovery* to let the ProActive Node find the URL to conn
 [[node_jar]]
 ==== Using node.jar
 
-It is also possible to launch a ProActive Node without even copying the ProActive Scheduler to a Compute Host:
+It is also possible to launch a ProActive Node without copying the ProActive Scheduler release to the Compute Host:
```
Comment (fviale) [inline src/docs/admin/ProActiveAdminGuide.adoc]: copying the ProActive Server release to

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 12]
```diff
@@ -585,9 +569,119 @@ The following changes must be performed:
    ...
 ----
 
-=== Deploy ProActive Nodes dynamically via other schedulers (PBS, SLURM, ...)
+[[_deploy_nodes_from_proactive_scheduler]]
+=== Deploy ProActive Nodes from the ProActive Scheduler
```
Comment (fviale) [inline src/docs/admin/ProActiveAdminGuide.adoc]: from the ProActive Resource Manager Interface

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 13]
```diff
@@ -585,9 +569,119 @@ The following changes must be performed:
...
+
+The second way of deploying ProActive Nodes is to create <<_glossary_node_source,*Node Sources*>> from the ProActive Scheduler.
```
Comment (fviale) [inline src/docs/admin/ProActiveAdminGuide.adoc]: same

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 14]
```diff
@@ -585,9 +569,119 @@ The following changes must be performed:
...
+
+ProActive Scheduler provides three ways to create a Node Source:
```
Comment (fviale) [inline src/docs/admin/ProActiveAdminGuide.adoc]: ProActive Resource Manager provides

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 15]
```diff
@@ -585,9 +569,119 @@ The following changes must be performed:
...
+The choice of *Infrastructure Manager* depends on the type of your Compute Hosts.
+The following part of this section presents how to deploy ProActive Nodes on various types of Compute Hosts:
+
+- <<_deploy_local_nodes,the local Server Host>>
+- <<_deploy_nodes_via_ssh,remote Compute Hosts accessible via SSH>>
+- <<_deploy_cloud_nodes,cloud platforms>>
+- <<_deploy_via_other_schedulers,clusters managed by other schedulers>>
+
+[[_deploy_local_nodes]]
+==== Deploy local nodes from the ProActive Scheduler
```
Comment (fviale) [inline src/docs/admin/ProActiveAdminGuide.adoc]: on the ProActive server

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 16]
```diff
@@ -585,9 +569,119 @@ The following changes must be performed:
...
+In order to deploy the ProActive nodes locally (i.e., on the <<_glossary_server_host,server host>> where running the ProActive Scheduler),
```
Comment (fviale) [inline src/docs/admin/ProActiveAdminGuide.adoc]: where the ProActive server is running

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 17]
```diff
@@ -585,9 +569,119 @@ The following changes must be performed:
...
+----
 
-This functionality is also called _*Meta-Scheduling*_.
+Then using this file create a Node Source either from the *Resource Manager Web Interface* or from the command line:
+[source]
+----
+$ PROACTIVE_HOME/bin/proactive-client -createns SSH_node_source --infrastructure org.ow2.proactive.resourcemanager.nodesource.infrastructure.SSHInfrastructure HOSTS_FILE 60000 3 5000 "" /home/user/jdk/bin/java PROACTIVE_HOME Linux "" config/authentication/rm.cred
+----
+
+Don't forget to replace `SCHEDULER_ADDRESS`, `HOSTS_FILE` and `PROACTIVE_HOME` with the corresponding values.
```
Comment (fviale) [inline src/docs/admin/ProActiveAdminGuide.adoc]: where is SCHEDULER_ADDRES used ? I don't see it in the command

File: src/docs/admin/ProActiveAdminGuide.adoc [hunk 18]
```diff
@@ -585,9 +569,119 @@ The following changes must be performed:
...
+
+See <<_ssh_infrastructure>> reference for details on each parameter.
+
+[[_deploy_cloud_nodes]]
+==== Deploy ProActive Nodes on Cloud Platforms
+
+The ProActive Scheduler has automated the deployment of ProActive Nodes on most popular Cloud Platforms.
```
Comment (fviale) [inline src/docs/admin/ProActiveAdminGuide.adoc]: The ProActive Resource Manager can automate the deployment ...

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc [hunk 19]
```diff
@@ -359,6 +359,68 @@ now be added using the *Create Node Source* panel in the Resource Manager or the
...
+-   *nodeTimeout*: The estimated startup time of the nodes (expressed in millisecond). After this timeout expired, the node is considered as lost.
+
+Using this configuration, you can start a Resource Manager and a Scheduler using the */bin/proactive-server* script.
+An GoogleComputeEngine NodeSource can now be added using the *Create Node Source* panel in the Resource Manager or the command line interface:
+
+    $ PROACTIVE_HOME/bin/proactive-client --createns googlecomputeengine -infrastructure org.ow2.proactive.resourcemanager.nodesource.infrastructure.GCEInfrastructure gceCredential totalNumberOfInstances numberOfNodesPerInstance vmUsername vmPublicKey vmPrivateKey image region ram cores rmHostname connectorIaasURL nodeJarURL additionalProperties nodeTimeout
+
+
+WARNING: When the Scheduler is stopped, the GoogleComputeEngine instances will be automatically deleted. And when the Scheduler is restarted, the infrastructure will be recovered as previous settings. The required GoogleComputeEngine instances will be re-created and re-configured.
```
Comment (fviale) [inline src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: When the ProActive server is stopped

File: src/docs/admin/ProActiveAdminGuide.adoc (hunk 5 above)
Comment (tao-xinxiu) [reply src/docs/admin/ProActiveAdminGuide.adoc]: Do you mean "familiar with ProActive Scheduler" -> "familiar with ProActive server configuration,"?
I think the second method (deploy nodes from the ProActive Scheduler) doesn't require the users to configure the ProActive Server. Did I misunderstand something?

Comment (fviale) [reply src/docs/admin/ProActiveAdminGuide.adoc]: familiar with ProActive scheduler is misleading, but I agree that familiar with ProActive server configuration is misleading also.
I suggest:
If you are not familiar with ProActive Workflows & Scheduling, ..

Comment (tao-xinxiu) [reply src/docs/admin/ProActiveAdminGuide.adoc]: Agree. I'll apply it.

Comment (tao-xinxiu) [reply src/docs/admin/ProActiveAdminGuide.adoc]: I change it to "If you are not familiar with ProActive Workflows & Scheduling product".

Comment (fviale) [reply src/docs/admin/ProActiveAdminGuide.adoc]: I like it

**Comment by fviale** [comment]
Retest this please

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc [hunk 20]
```diff
@@ -212,7 +212,7 @@ now be added using the *Create Node Source* panel in the Resource Manager or the
 
 As AWS is a paying service, when the Scheduler is stopped normally (without removing the created infrastructure), all the created AWS instances will be terminated. And when the Scheduler is restarted, these instances will be re-configured as per previous settings.
 
-WARNING: If the Scheduler is forced killed, the created AWS instances will not be terminated. And, when the Scheduler is restarted, the infrastructure will be re-configured as per previous settings. If the instances were delete at the AWS side, they will be re-created and re-configured.
+WARNING: If the ProActive server is forced killed, the created AWS instances will not be terminated. And, when the ProActive server is restarted, the infrastructure will be re-configured as per previous settings. If the instances were delete at the AWS side, they will be re-created and re-configured.
```
Comment (nebilbenmabrouk) [inline src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: - ProActive server  is **forcibly killed**
- be re-configured as per **the** previous settings.
- If the instances were **deleted** at

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc [hunk 21]
```diff
@@ -292,7 +292,7 @@ now be added using the *Create Node Source* panel in the Resource Manager or the
     $ PROACTIVE_HOME/bin/proactive-client --createns openstack -infrastructure org.ow2.proactive.resourcemanager.nodesource.infrastructure.OpenstackInfrastructure username password endpoint rmHostname connectorIaasURL image flavor publicKeyName numberOfInstances numberOfNodesPerInstance downloadCommand additionalProperties
 
 
-WARNING: When the Scheduler is stopped (without removing the created infrastructure), the Openstack instances will not be terminated. And when the Scheduler is restarted, the infrastrucutre will be re-configured as per previous settings. If the instances were delete at the Openstack server side, they will be re-created and re-configured.
+WARNING: When the ProActive server is stopped (without removing the created infrastructure), the Openstack instances will not be terminated. And when the ProActive server is restarted, the infrastrucutre will be re-configured as per previous settings. If the instances were delete at the Openstack server side, they will be re-created and re-configured.
```
Comment (nebilbenmabrouk) [inline src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: - When ProActive server is stopped (without removing the created infrastructure), Openstack instances 
- when ProActive server is restarted, 
- will be re-configured as per **the** previous settings. 
-If the instances were **deleted** at the Openstack **Cloud** side,

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc [hunk 22]
```diff
@@ -357,8 +357,70 @@ now be added using the *Create Node Source* panel in the Resource Manager or the
     $ PROACTIVE_HOME/bin/proactive-client --createns vmware -infrastructure org.ow2.proactive.resourcemanager.nodesource.infrastructure.VmwareInfrastructure username password endpoint rmHostname connectorIaasURL image ram cores vmusername vmpassword numberOfInstances numberOfNodesPerInstance downloadCommand additionalProperties
 
 
-WARNING: When the Scheduler is stopped (without removing the created infrastructure), the VmWare instances will not be terminated. And when the Scheduler is restarted, the infrastrucutre will be re-configured as per previous settings. If the instances were delete at the VmWare server side, they will be re-created and re-configured.
+WARNING: When the ProActive server is stopped (without removing the created infrastructure), the VmWare instances will not be terminated. And when the ProActive server is restarted, the infrastrucutre will be re-configured as per previous settings. If the instances were delete at the VmWare server side, they will be re-created and re-configured.
```
Comment (nebilbenmabrouk) [inline src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: Please replace 'the ProActive server' simply with 'ProActive server' in the whole doc
replace  remove **the**  in sentences such as 'the VmWare instances' in the whole doc
add **the** before 'previous settings'.
replace **delete** with **deleted** in the whole doc

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc (hunk 20 above)
Comment (tao-xinxiu) [reply src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: A small point to confirm, "be re-configured as **per** the previous settings." should be "be re-configured as the previous settings.", right?

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc [hunk 23]
```diff
@@ -357,8 +357,70 @@ now be added using the *Create Node Source* panel in the Resource Manager or the
...
+4. Under the tab _Credentials_, Click _Create credentials_ > _Service account key_.
+5. Select the option _New service account_, then fill in your service account name, choose _Compute Engine_ > _Compute Admin_ as "role", choose _JSON_ as "Key type".
+6. Click the button _Create_, a JSON file for the created service account will be downloaded.
+
+For more information regarding Google Cloud service accounts see
+https://cloud.google.com/compute/docs/access/service-accounts[Google Cloud Service Accounts].
+
+
+Now, you are ready to create a new node source of the type _GCE Infrastructure_.
+The downloaded JSON file is used to fill in the properties *gceCredential* to perform the basic Google Cloud Platform authentification.
```
Comment (nebilbenmabrouk) [inline src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: which JSON file?

File: src/docs/admin/references/NodeSourcePolicyReference.adoc [hunk 24]
```diff
@@ -451,3 +451,15 @@ node and deploying another one whereas the task could have been
 scheduled on the same node. This releaseDelay therefore represents the
 time to wait before effectively removing a node.
 
+====== Native Scheduler Policy
+
+The Native Scheduler Policy interacts with the <<_glossary_ns_native_scheduler,native scheduler>> to request ProActive Nodes deployment dynamically based on the <<_glossary_proactive_scheduler,*ProActive Scheduler*>> pending queue.
```
Comment (nebilbenmabrouk) [inline src/docs/admin/references/NodeSourcePolicyReference.adoc]: ProActive **nodes**

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc (hunk 20 above)
Comment (nebilbenmabrouk) [reply src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: To me 'be re-configured as per the previous settings' is better. I'll let you choose the proper sentence according to you.

File: src/docs/admin/references/NodeSourceInfrastructureReference.adoc (hunk 23 above)
Comment (tao-xinxiu) [reply src/docs/admin/references/NodeSourceInfrastructureReference.adoc]: Changed to "The downloaded JSON file **of Google Cloud credentials** is used to..."

**Pull request merged**